add quantized_resize and dequantize for some cuda backends

hyuen opened the pull request:
Test Plan: added tests with the diff, verifying the windows failures

Differential Revision: D29494134

hyuen commented:
The failure in windows is environmental, it failed to install vscode

supriyar commented on aten/src/ATen/native/native_functions.yaml:
```diff
@@ -7243,6 +7245,7 @@
     CPU: cpu_equal
     CUDA: cuda_equal
     QuantizedCPU: equal_quantized_cpu
+    QuantizedCUDA: cuda_equal_quantized
```
should we name this `equal_quantized_cuda` to be consistent with above?

supriyar commented on aten/src/ATen/native/quantized/cuda/base_functions.cu:
```diff
@@ -0,0 +1,34 @@
+namespace at {
+namespace native {
+
+bool cuda_equal_quantized(const Tensor& self, const Tensor& src) {
```
Are we checking for equality of quantized tensors here? I think we should also check the qscheme and scale/zero_point in that case.
cc @jerryzh168 looked at something similar for all_close for quantized tensors

supriyar commented on test/quantization/core/test_quantized_tensor.py:
```diff
@@ -864,6 +864,66 @@ def test_qtensor_resize(self):
             q4 = q1.contiguous().resize(*sizes2)
 
+    @unittest.skipIf(not torch.cuda.is_available() or TEST_WITH_ROCM, 'CUDA is not available')
+    def test_cuda_cpu_qimplementation_consistency(self):
```
Is this a general test for testing quantize_per_tensor/dequantize on cpu/cuda? It doesn't seem related to the functions added above, unless I'm missing something

supriyar commented on test/quantization/core/test_quantized_tensor.py:
```diff
@@ -864,6 +864,66 @@ def test_qtensor_resize(self):
+        q3 = q1.resize(*sizes2)
+        q4 = q1.contiguous().resize(*sizes2)
+
+
```
Can we also add test for `equal` and `empty_strided`?

Commit:
add quantized_resize and dequantize for some cuda backends (#61057)

Summary: Pull Request resolved: https://github.com/pytorch/pytorch/pull/61057

Test Plan: added tests with the diff, verifying the windows failures

Differential Revision: D29494134

fbshipit-source-id: 25641fb9b3293ae0dd605a3fa6b0536a85e14293

jerryzh168 commented on aten/src/ATen/native/native_functions.yaml:
```diff
@@ -1796,7 +1796,7 @@
     CPU, Meta: resize_
     CUDA: resize_cuda_
-    QuantizedCPU: quantized_resize_cpu_
+    QuantizedCPU, QuantizedCUDA: quantized_resize_cpu_
```
following naming convention for quantized functions, this should be resize_quantized_ I think

jerryzh168 commented on aten/src/ATen/native/native_functions.yaml:
```diff
@@ -1816,6 +1816,7 @@
   dispatch:
     CPU: empty_strided_cpu
     CUDA: empty_strided_cuda
+    QuantizedCUDA: empty_strided_cuda_quantized
```
nit: empty_strided_quantized_cuda

the naming convention is function_name_{backend_name}
